### ISSUE TITLE
MSX system hangs on PSG manipulation

### ISSUE THREAD

**Issue by apoloval** [apoloval/artemisa#94]
Some hangs of the MSX system detected during #93. 

How to reproduce it:

- Execute these statements in MSX Basic:
```
10 A$="t100l8o4gbo5d4o4go5ce4"
20 PLAY A$
RUN
30 END
```
- The system hangs

This does not work both in the Artemisa and in OpenMSX emulator.

**Comment by apoloval** [comment]
Launching the debugger in OpenMSX I realized the hang is caused by configuring slot 0 (ROM) for page 3. This page is RAM, and must be RAM all the time. Otherwise, we would loose the Basic working area, the stack, etc. This is a failure, indeed.

The following stack trace can be obtained from the debugger. 
```
RDPRIM  $F380
RDSLT   $000C
  BIOS call: read from slot
  Args:
    HL (address to read): $C006
    A (slot selection): $00 (primary 0, secondary 0)
C7E1A   $7E1A
  Subroutine: read word from extension ROM
  Args:
    HL (address): $C006
    C (slot ID): $00
  Calls RDSLT from $7E21
J564A   $564A
  Subroutine:  i/o function dispatcher for expansion ROM
  Calls C7E2D to translate SLTATR entrynumber to address and slotid
  Calls C7E1A from $5657
C6F8F    $6F8F
  Subroutine:  i/o function dispatcher
  Jumps to J564A if i/o channel device is not a internal device
```

In summary, it seems we have a problem in this code:
```asm
C6F8F:  CALL    H.GEND
        PUSH    HL
        PUSH    DE
        PUSH    AF
        LD      DE,4
        ADD     HL,DE
        LD      A,(HL)
        CP      0FCH                    ; i/o channel device a internal device ?
        JP      C,J564A                 ; nope, start i/o function in expansion ROM
```

**Comment by apoloval** [comment]
Finally I tried with a different approach. Check the differences in the hex dumps between our ROM image and the SVI-728 ROM. Byte after byte, I detected and fixed [this problem with a constant wrongly generated by Pasmo](https://github.com/apoloval/msx-system/commit/78b00d22906773738e3d6f114ece3b7f4ac67214). Once fixed, the bug is no longer reproduced in OpenMSX. 

Leaving this issue opened until we test in a real Artemisa machine.
